src/compiler/checker.ts

```diff
@@ -6577,6 +6577,21 @@ namespace ts {
             let container = getThisContainer(node, /* includeArrowFunctions */ true);
             let needToCaptureLexicalThis = false;
 
+

```

Comment:
I agree with you though I think that will need to be a separate change. Here is a issue #5232